### PR TITLE
Lobby Attention Buttons

### PR DESCRIPTION
**Pull request by Hughgent**
## About The Pull Request

Saw people arguing about medbay. Came up with this idea.

Could also create a version for requisitions and engineering.

## Why It's Good For The Game

Allows doctors to be elsewhere and get notified when there are patients waiting for them. Could also be scattered about the ship allowing for people to ping a doctor directly rather than screaming over general comms.

## Changelog
:cl: Hughgent
add: Lobby notification button for medbay.
/:cl:

### CODE REVIEW

File: code/game/machinery/buttons.dm [hunk 1]
```diff
@@ -47,4 +47,37 @@
...
 	attack_paw(mob/user as mob)
-		return
\ No newline at end of file
+		return
+
+/obj/machinery/medical_help_button
+	name = "Medical attention required"
+	icon = 'icons/obj/stationobjs.dmi'
+	icon_state = "doorctrl0"
+	desc = "A button for alerting doctors that you require assitance."
```
Comment (Shadow-Quill) [inline code/game/machinery/buttons.dm]:
```suggestion
	desc = "A button for alerting doctors that you require assistance."
```
Muh spelling

Comment (Hughgent) [reply code/game/machinery/buttons.dm]: I type fast.

File: code/game/machinery/buttons.dm [hunk 2]
```diff
@@ -47,4 +47,37 @@
...
+	use_power(5)
+	icon_state = "doorctrl1"
+	add_fingerprint(user)
+
+	var/mob/living/silicon/ai/AI = new/mob/living/silicon/ai(src, null, null, 1)
+	AI.SetName("Lobby Notification System")
+	AI.aiRadio.talk_into(AI,"<b>[user.name] is requesting medical attention at: [get_area(src)].</b>","MedSci","announces")
+	qdel(AI)	
+
+	spawn(15)
```
Comment (Rohesie) [inline code/game/machinery/buttons.dm]: Use a CALLBACK here.

**Comment by Hughgent** [comment]
image

**Comment by Hughgent** [comment]
image

File: code/game/machinery/buttons.dm [hunk 3]
```diff
@@ -47,4 +47,50 @@
...
+	active = TRUE
+	addtimer(CALLBACK(src, .proc/icon_update_check), 10 SECONDS)
+
+/obj/machinery/medical_help_button/proc/icon_update_check()
+	active = FALSE
+	if(!(stat & NOPOWER))
+		icon_state = "doorctrl0"
+
+/obj/machinery/medical_help_button/power_change()
+	..()
```
Comment (Rohesie) [inline code/game/machinery/buttons.dm]:
```suggestion
	. = ..()
```

**Comment by Hughgent** [comment]
also requires mapping

**Comment by Rohesie** [comment]
Travis check failing.

File: code/game/machinery/buttons.dm [hunk 4]
```diff
@@ -47,4 +47,50 @@
...
+	desc = "A button for alerting doctors that you require assistance."
+	var/id = null
+	var/active = FALSE
+	anchored = 1.0
+	use_power = 1
+	idle_power_usage = 2
+	active_power_usage = 4
+
+/obj/machinery/medical_help_button/attack_hand(mob/living/carbon/human/user)
+	add_fingerprint(user)
```
Comment (DominikPanic) [inline code/game/machinery/buttons.dm]: This should likely be below the istype. (as if they can't use it anyway what use is to know for us that they touched it? Do xenos even have fingerprints?)

File: code/game/machinery/buttons.dm [hunk 5]
```diff
@@ -47,4 +47,50 @@
...
-		return
\ No newline at end of file
+		return
+
+/obj/machinery/medical_help_button
+	name = "Medical attention required"
+	icon = 'icons/obj/stationobjs.dmi'
+	icon_state = "doorctrl0"
+	desc = "A button for alerting doctors that you require assistance."
+	var/id = null
```
Comment (DominikPanic) [inline code/game/machinery/buttons.dm]: Why define it if you aren't using it in any way?

Comment (Hughgent) [reply code/game/machinery/buttons.dm]: copy paste from other working buttons.

**Pull request merged**